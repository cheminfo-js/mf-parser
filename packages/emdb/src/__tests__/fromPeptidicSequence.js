'use strict';

const DBManager = require('..');

test('test fromArray', () => {
    let dbManager = new DBManager();
    dbManager.fromPeptidicSequence('AAKK', {
        allowNeutralLoss: false,
        protonation: false,
        protonationPH: 7,
        ionizations: 'H+,Na+',
        fragmentation: {
            a: true
        },
        mfFilter: {
            minMSEM: 100,
            maxMSEM: 300
        },
    });

    expect(dbManager.databases.peptidic).toHaveLength(2);
    expect(dbManager.databases.peptidic).toMatchSnapshot();
});
