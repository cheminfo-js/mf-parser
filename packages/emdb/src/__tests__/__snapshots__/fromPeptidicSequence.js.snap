// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test fromPeptidicSequence 1`] = `
Array [
  Object {
    "atoms": Object {
      "C": 11,
      "H": 23,
      "N": 4,
      "O": 2,
    },
    "charge": 1,
    "comment": "a3",
    "em": 243.18210099815,
    "ionization": Object {
      "charge": 1,
      "em": 1.00782503223,
      "mf": "H+",
    },
    "mf": "C11H23N4O2(+1)",
    "ms": Object {
      "charge": 2,
      "delta": -27.905585564719075,
      "em": 122.09441443528092,
      "ionization": "H+",
      "ppm": 186037.23709812717,
    },
    "mw": 243.32635490179035,
    "parts": Array [
      "HAlaAlaLysC-1O-1(+1)",
    ],
    "unsaturation": 2,
  },
  Object {
    "atoms": Object {
      "C": 11,
      "H": 23,
      "N": 4,
      "O": 2,
    },
    "charge": 1,
    "comment": "a3",
    "em": 243.18210099815,
    "ionization": Object {
      "charge": 1,
      "em": 22.989769282,
      "mf": "Na+",
    },
    "mf": "C11H23N4O2(+1)",
    "ms": Object {
      "charge": 2,
      "delta": -16.914613439834056,
      "em": 133.08538656016594,
      "ionization": "Na+",
      "ppm": 112764.08959889371,
    },
    "mw": 243.32635490179035,
    "parts": Array [
      "HAlaAlaLysC-1O-1(+1)",
    ],
    "unsaturation": 2,
  },
]
`;
